Kwf/User/Auth/ActivationFields.php

```diff
@@ -6,35 +6,14 @@
  */
 class Kwf_User_Auth_ActivationFields extends Kwf_User_Auth_Abstract implements Kwf_User_Auth_Interface_Activation
 {
-    private function _encodePasswordBcrypt(Kwf_Model_Row_Interface $row, $password)
-    {
-        $rounds = '08';
-        $string = $this->_getHashHmacStringForBCrypt($row, $password);
-        $salt = substr ( str_shuffle ( './0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ' ) , 0, 22 );
-        return crypt ( $string, '$2a$' . $rounds . '$' . $salt );
-    }
-
-    private function _getHashHmacStringForBCrypt(Kwf_Model_Row_Interface $row, $password)
-    {
-        $globalSalt = Kwf_Registry::get('config')->user->passwordSalt;
-        return hash_hmac ( "whirlpool", str_pad ( $password, strlen ( $password ) * 4, sha1 ( $row->id ), STR_PAD_BOTH ), $globalSalt, true );
-    }
-
-    private function _validateActivateTokenBcrypt($row, $token)
-    {
-        $activateToken = explode(':', $row->activate_token);
-        $string = $this->_getHashHmacStringForBCrypt($row, $token);
-        return crypt($string, substr($activateToken[1], 0, 30));
-    }
-
     public function validateActivationToken(Kwf_Model_Row_Interface $row, $token)
     {
         if (!$row->activate_token) return false;
         $activateToken = explode(':', $row->activate_token);
         $expire = $activateToken[0];
         $rowToken = $activateToken[1];
         if ($expire < time()) return false;
-        if ($this->_validateActivateTokenBcrypt($row, $token) === $rowToken) {
+        if (Kwf_User_Auth_Bcrypt::validateValue($row, $token, $rowToken) === $rowToken) {
```

Comment:
rowToken ist eh in row warum hier als extra argument?

Reply:
@kaufmo weils unterschiedlich verwendet wird. Es gibt die passwort-version, da wird auf $row->password geschaut, und die token version, da wird $row->activate_token verwendet.